rpcs3/Emu/Cell/Modules/cellSaveData.cpp

```diff
@@ -141,8 +141,15 @@ static NEVER_INLINE s32 savedata_op(ppu_thread& ppu, u32 operation, u32 version,
 						save_entry2.atime = entry.atime;
 						save_entry2.mtime = entry.mtime;
 						save_entry2.ctime = entry.ctime;
-						//save_entry2.iconBuf = NULL; // TODO: Here should be the PNG buffer
-						//save_entry2.iconBufSize = 0; // TODO: Size of the PNG file
+						if (fs::is_file(base_dir + entry.name + "/ICON0.PNG"))
+						{
+							fs::file icon = fs::file(base_dir + entry.name + "/ICON0.PNG");
+							u32 iconSize = icon.size();
+							uchar* iconData = new uchar[iconSize];
+							icon.read(iconData, iconSize);
```

Comment:
True. I wouldn't have to deal with the std::shared_ptr<void> either since the vector would copy and deallocate automatically as needed. Tbh I only used the void* pointer because initially the comment used it as such.